miio/walkingpad.py

```diff
@@ -0,0 +1,280 @@
+    def stop(self):
+        """Stop the treadmill."""
+        return self.send("set_state", ["stop"])
+
+    @command(
+        click.argument("mode", type=EnumType(OperationMode)),
+        default_output=format_output("Setting mode to '{mode.name}'"),
+    )
+    def set_mode(self, mode: OperationMode):
+        """Set mode (auto/manual)."""
+
+        if not isinstance(mode, OperationMode):
+            raise WalkingpadException("Invalid mode: %s" % mode)
+
+        return self.send("set_mode", [mode.value])
+
+    @command(
+        click.argument("speed", type=float),
+        default_output=format_output("Setting speed to {speed}"),
+    )
+    def set_speed(self, speed: float):
+        """Set speed."""
+
+        if not isinstance(speed, float):
+            raise WalkingpadException("Invalid speed: %s" % speed)
+
+        if speed < 0 or speed > 6:
+            raise WalkingpadException("Invalid speed: %s" % speed)
+
+        return self.send("set_speed", [speed])
+
+    @command(
+        click.argument("speed", type=float),
+        default_output=format_output("Setting start speed to {speed}"),
+    )
+    def set_start_speed(self, speed: float):
+        """Set start speed."""
+
+        if not isinstance(speed, float):
+            raise WalkingpadException("Invalid start speed: %s" % speed)
+
+        if speed < 0 or speed > 6:
+            raise WalkingpadException("Invalid start speed: %s" % speed)
+
+        return self.send("set_start_speed", [speed])
+
+    @command(
+        click.argument("sensitivity", type=EnumType(OperationSensitivity)),
+        default_output=format_output("Setting sensitivity to {sensitivity}"),
+    )
+    def set_sensitivity(self, sensitivity: OperationSensitivity):
+        """Set sensitivity."""
+
+        if not isinstance(sensitivity, OperationSensitivity):
+            raise WalkingpadException("Invalid mode: %s" % sensitivity)
+
+        return self.send("set_sensitivity", [sensitivity.value])
+
+    def _get_quick_status(self):
+
+        # internal helper to get the quick status via the "all" property
+
+        properties = ["all"]
+
+        values = self.get_properties(properties, max_properties=1)
+
+        data = {}
+        for x in values:
+            prop, value = x.split(":")
+
+            if prop not in ["sp", "step", "cal", "time", "dist", "mode"]:
+                raise WalkingpadException("Invalid data received: %s" % value)
+
+            # Ensure that he different properties are correctly typed
+            if prop == "sp":
+                value = float(value)
+            elif prop == "step":
+                value = int(value)
+            elif prop == "cal":
+                value = int(value)
+            elif prop == "time":
+                value = int(value)
+            elif prop == "dist":
+                value = int(value)
+            elif prop == "mode":
+                value = int(value)
+
+            data[prop] = value
```

Comment:
Something like this would be more readable:
```
# prepare data dict
data = ...

value_map = {
    "time": int,
    "mode": int,
    "sp": float,
    ...
}

converted_data = {key: type_map[key](value) for key, value in data.items()}
```

Reply:
Same here! Really appreciate the feedback and help... learning a lot!